### PR TITLE
Gantt Chart / Service Map followup

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
Fast follow action items on code styling from:
https://github.com/opensearch-project/dashboards-observability/pull/2257
https://github.com/opensearch-project/dashboards-observability/pull/2264
Moving functions declaration out of use effect
Styling change for onRelayout
Cypress test flakiness fixes

Bug fix: resizing handling for Span list / Tree view
Before:

https://github.com/user-attachments/assets/8c443d21-2619-4056-aa9c-7a3124eaf594

After: (Now reacts correctly to full screen and side navigation changes)

https://github.com/user-attachments/assets/138d0727-e36c-489a-9aca-844d53db94fe

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js [hunk 1]
```diff
@@ -201,7 +201,8 @@ describe('Testing Service map', () => {
     cy.get('[data-text="Errors"]').click();
     cy.contains('60%');
     cy.get('[data-text="Duration"]').click();
-    cy.contains('100');
```
Comment (paulstn) [inline .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js:204]: Is it okay to not check for this Duration value anymore?

File: .cypress/integration/trace_analytics_test/trace_analytics_traces.spec.js [hunk 2]
```diff
@@ -125,32 +125,34 @@ describe('Testing traces table', () => {
     setTimeFilter();
   });
 
-  it('Renders the traces table and verify Table Column, Pagination and Rows Data ', () => {
+  it.only('Renders the traces table and verify Table Column, Pagination and Rows Data ', () => {
```
Comment (paulstn) [inline .cypress/integration/trace_analytics_test/trace_analytics_traces.spec.js:128]: I believe `.only` will only run this test in the suite

File: .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js (hunk 1 above)
Comment (TackAdam) [reply .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js:204]: The duration value was just reading the 100 off the sidebar, cy.contains('Average duration (ms)'); is a more accurate check if switching the metrics worked. This is just to cover the rendering to replace the wait functionality though as this test is for the node details checked on line 210

cy.get('.euiText.euiText--small').contains('Average duration: 216.43ms').should('exist');

File: .cypress/integration/trace_analytics_test/trace_analytics_traces.spec.js (hunk 2 above)
Comment (TackAdam) [reply .cypress/integration/trace_analytics_test/trace_analytics_traces.spec.js:128]: Good catch thank you forgot to remove

File: .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js (hunk 1 above)
Comment (paulstn) [reply .cypress/integration/trace_analytics_test/trace_analytics_services.spec.js:204]: Thanks, sounds good to me!

**Review by paulstn — APPROVED** [review]
LGTM

**Review by mengweieric — APPROVED** [review]

**Pull request merged**